pilot/pkg/config/clusterregistry/secretcontroller.go

```diff
@@ -244,31 +246,32 @@ func addMemberCluster(s *corev1.Secret, c *Controller) {
 				Controller:       kubectl,
 			})
 		stopCh := make(chan struct{})
+		c.cs.rc[key].ControlChannel = stopCh
```

Comment:
If you look at the stop channel for the existing controllers, the go routine associated with the controller terminates when the stop channel is closed. See the WaitSignal() call in pilot-discovery/main.go. I think that that logic should also apply to the controllers dynamically created for multi-cluster (in addition to this additional stopch created at line 249).

Reply:
As I mentioned above, this is a standard way of informing go routine to stop via stop channel. Please try it and you will see it works as expected.